### PR TITLE
feat(@desktop/wallet): Use SubModelProxyModel in SwapAdaptor.qml

### LINKED ISSUE #14926: Use SubModelProxyModel in SwapAdaptor.qml
### Description

Following up on the todo added https://github.com/status-im/status-desktop/pull/14900#pullrequestreview-2080552879 need to implement this

### PR DESCRIPTION
**Pull request by Khushboo-dev-cpp**
fixes #14926

### What does the PR do

Uses the SubModelProxyModel after fix from @micieslak https://github.com/status-im/status-desktop/pull/14900#pullrequestreview-2080552879

### Affected areas

SwapModal

### StatusQ checklist

- [ ] add documentation if necessary (new component, new feature)
- [ ] update sandbox app
  - in case of new component, add new component page
  - in case of new features, add variation to existing component page
  - nice to have: add it to the demo application as well
- [ ] test changes in both light and dark theme?

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

### CODE REVIEW

File: ui/app/AppLayouts/Wallet/popups/swap/SwapModalAdaptor.qml [hunk 1]
```diff
@@ -32,8 +32,11 @@ QObject {
                 expectedRoles: ["address"]
             },
             FastExpressionRole {
+                function getFromToken() {
+                    return ModelUtils.getByKey(root.walletAssetsStore.walletTokensStore.plainTokensBySymbolModel, "key", root.swapFormData.fromTokensKey)
```
Comment (stefandunca) [inline ui/app/AppLayouts/Wallet/popups/swap/SwapModalAdaptor.qml:36]: This value is not depending on `model` data and it is the same static value. Maybe it makes sense to be cached on `root.swapFormData.fromTokensKey` change

**Review by stefandunca — APPROVED** [review]
👍

File: ui/app/AppLayouts/Wallet/popups/swap/SwapModalAdaptor.qml (hunk 1 above)
Comment (caybro) [reply ui/app/AppLayouts/Wallet/popups/swap/SwapModalAdaptor.qml:36]: Yup, the extra roles are not cheap

**Review by caybro — APPROVED** [review]
Otherwise LGTM

**Pull request merged**